Cython/Compiler/PyrexTypes.py

```diff
@@ -695,9 +695,9 @@ def is_complete(self):
     def declaration_code(self, entity_code,
             for_display = 0, dll_linkage = None, pyrex = 0):
         # XXX: we put these guards in for now...
-        assert not pyrex
         assert not dll_linkage
         from . import MemoryView
+        for_display = for_display or pyrex
         base_code = StringEncoding.EncodedString(
                         (str(self)) if for_display else MemoryView.memviewslice_cname)
```

Comment:
Let's avoid adding to the semantic ambiguity between `pyrex` and `for_display`.
```suggestion
        base_code = StringEncoding.EncodedString(
                        (str(self)) if pyrex or for_display else MemoryView.memviewslice_cname)
```